lib/logstash/inputs/tcp.rb

```diff
@@ -260,14 +260,24 @@ def handle_socket(socket)
   end
 
   def enqueue_decorated(event, client_ip_address, client_address, client_port, socket)
-    event.set(HOST_FIELD, client_address) unless event.get(HOST_FIELD)
-    event.set(HOST_IP_FIELD, client_ip_address) unless event.get(HOST_IP_FIELD)
-    event.set(PORT_FIELD, client_port) unless event.get(PORT_FIELD)
-    event.set(SSLSUBJECT_FIELD, socket.peer_cert.subject.to_s) if socket && @ssl_enable && @ssl_verify && event.get(SSLSUBJECT_FIELD).nil?
+    event.set(@field_host, client_address) unless event.get(@field_host)
+    event.set(@field_host_ip, client_ip_address) unless event.get(@field_host_ip)
+    event.set(@field_port, client_port) unless event.get(@field_port)
+    event.set(@field_sslsubject, socket.peer_cert.subject.to_s) if socket && @ssl_enable && @ssl_verify && event.get(@field_sslsubject).nil?
     decorate(event)
     @output_queue << event
   end
 
+  # setup the field names, with respect to ECS compatibility.
+  def setup_fields!
+    @field_host       = ecs_select[disabled: "host",                    v1: "[@metadata][tcp][source][name]"        ].freeze
```

Comment:
in plugins we often use a naming convention for the meta-data e.g. 
- `[@metadata][input][beats][tls][client][subject]`
- or `[@metadata][input][http][request][headers]`

there isn't a tcp filter to conflict here but would still be nice to be explicit using the `[input][tcp]` prefix, wdyt?

we do not follow the convention everywhere , for me personally it would make sense from a user perspective
e.g. having a bare `[@metadata][path]` in file input vs having `[@metadata][input][file][path]`